front-end/src/app/app.component.spec.ts

```diff
@@ -1,12 +1,14 @@
 import { TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
 import { AppComponent } from './app.component';
+import {HttpClientModule} from '@angular/common/http';
 
 describe('AppComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [
-        RouterTestingModule
+        RouterTestingModule,
+        HttpClientModule,
```

Comment:
Remove HttpClientModule and use a mock for your service 

Lock here

https://angular.io/guide/testing-services#:~:text=When%20testing%20a,a%20spy%20object.